modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/partials/customized.jsp

```diff
@@ -27,5 +27,6 @@ FDSSampleDisplayContext fdsSampleDisplayContext = (FDSSampleDisplayContext)reque
 	propsTransformer="{SampleFDSPropsTransformer} from frontend-data-set-sample-web"
 	selectedItemsKey="id"
 	selectionType="multiple"
+	showBulkActionsManagementBarActions="<%= false %>"
```

Comment:
Looking at https://github.com/liferay-frontend/liferay-portal/pull/4370#issuecomment-2285583475, I realize my suggestion was waay too vague.

What we want are three tests for new behaviors:
1. `onSelect` was invoked, check results. IIUC, this has nothing to do with two new flags. We can set this up on `customized` sample, witch typically has most complicated code.
2. Test the effect of `showBulkActionsManagementBar` being set to `false`. We can do this on any of the samples. I'd avoid `customized`, as it has the biggest set of features, so it feels natural for it to have the management bar. I'd also avoid `minimal` and `empty`, for obvious reasons. That leaves two, `react` or `controlled`. It would need to have selection enabled.
3. Test the effect of `showBulkActionsManagementBarActions` being set to `false`. We can do this on the remaining sample. It would also need to have selection enabled.

We also do need these tests in playwright. We don't want to rely on commerce for test coverage for FDS. That's the point of adding sample behaviors to the sample module. Without tests, there is no point to even add them. If you need help with tests, let us know and we can add it to our planned work cc @dsanz @herrardo

Reply:
Super clear now, thanks @markocikos!

And of course it does make sense to add tests coincidentally with the new features of course, but it'll be hard for me to deliver them as I'm on a tight schedule and there's also a dependency on this PR. Could please I ask for some help from one of you guys?

Thanks in advance!

Reply:
After a conversation with @gianmarcobrunialti and @juanjofgliferay, we concluded that:

1. Purpose (use case) of the `onItemSelected` makes sense on its own: having a callback receiving currently selected items allows implementing "external bulk actions" that are not necessarily tied to the FDS UI
2. Nevertheless, this new piece of API is very close to the existing `onSelect()` callback, therefore, this must be reviewed to unify selection management and exposure into a single abstraction or two, well-distinguished ones
3. In addition, proper testing for the new features is a must in an application-independent way, following the traditional pattern of adding the feature usage to the sample + extending PW tests targeting it. Strategy outlined by @markocikos [here](https://github.com/liferay-frontend/liferay-portal/pull/4370#discussion_r1768443456) heads in the right direction
4. Timeframe for commitment of this is tight

As a result of the above, here is the plan we made:

- We'll let this PR go in its current form
- We'll schedule time to deal with tests here → https://liferay.atlassian.net/browse/LPD-37357
- We'll schedule time to analyze current selection API and attempt to simplify it here → https://liferay.atlassian.net/browse/LPD-37358

Both of the above must be completed in the next quarter, that's why I've added them to the [Data Set Quality Improvements](https://liferay.atlassian.net/browse/LPD-34637) epic
cc/ @herrardo